### PR TITLE
TTWEBTAKEN-134: Add php 8 support

### PR DESCRIPTION
**Pull request by MPParsley**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the CHANGELOG accordingly.
- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
             "email": "peter.decuyper@digipolis.gent"
         }
     ],
-    "minimum-stability": "dev",
-    "prefer-stable": true,
+    "homepage": "https://bitbucket.org/digipolisgent/php_package_gent-services-profile",
     "require": {
+        "php": "^7.4 || ^8.1",
+        "digipolisgent/api-client": "^1.1.0",
+        "digipolisgent/value": "^1.1",
```
Comment (zero2one) [inline composer.json:22]: Both need to be upgraded to 3.0 version

File: composer.json [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
-        "php": ">=7.2",
-        "psr/simple-cache": "^1.0",
-        "digipolisgent/value": "^1.1",
-        "digipolisgent/api-client": "^1.1.0"
+        "psr/simple-cache": "^1.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^8",
-        "squizlabs/php_codesniffer": "^2.3",
         "brainmaestro/composer-git-hooks": "^2.4",
```
Comment (zero2one) [inline composer.json]: This can be removed.

File: composer.json [hunk 3]
```diff
@@ -1,32 +1,36 @@
...
-        "mihaeu/html-formatter": "*"
+        "district09/qa-php": "^1.0.1",
+        "mihaeu/html-formatter": "*",
+        "phpunit/phpunit": "^9"
```
Comment (zero2one) [inline composer.json]: This line can be removed as it is included by`district09/qa-php`

File: composer.json [hunk 4]
```diff
@@ -1,32 +1,36 @@
...
         }
     ],
-    "minimum-stability": "dev",
-    "prefer-stable": true,
+    "homepage": "https://bitbucket.org/digipolisgent/php_package_gent-services-profile",
     "require": {
+        "php": "^7.4 || ^8.1",
+        "digipolisgent/api-client": "^1.1.0",
+        "digipolisgent/value": "^1.1",
         "guzzlehttp/guzzle": "^6.2",
```
Comment (zero2one) [inline composer.json:23]: Guzzle is provided by the `digipolisgent/api-client:^3.0 version.

File: composer.json [hunk 5]
```diff
@@ -1,32 +1,36 @@
...
             "name": "Peter Decuyper",
             "email": "peter.decuyper@digipolis.gent"
         }
     ],
-    "minimum-stability": "dev",
-    "prefer-stable": true,
+    "homepage": "https://bitbucket.org/digipolisgent/php_package_gent-services-profile",
     "require": {
+        "php": "^7.4 || ^8.1",
+        "digipolisgent/api-client": "^1.1.0",
```
Comment (zero2one) [inline composer.json:21]: Add missing `"ext-json": "*"`.

File: composer.json [hunk 6]
```diff
@@ -37,23 +41,29 @@
...
         ],
-        "post-install-cmd": [
-            "vendor/bin/cghooks update"
-        ]
-    },
-    "hooks": {
-        "pre-commit": [
-            "vendor/bin/phpcs -p --standard=PSR2 --runtime-set ignore_errors_on_exit 1 --runtime-set ignore_warnings_on_exit 1 examples src tests\n"
+        "test": [
+            "vendor/bin/phpunit"
```
Comment (zero2one) [inline composer.json]: The scripts should be replaced by:

```json
    "scripts": {
        "post-install-cmd": "vendor/bin/grumphp git:init",
        "coverage": "vendor/bin/phpunit --configuration=phpunit.qa-php.xml --coverage-html build/coverage",
        "grumphp": "vendor/bin/grumphp run",
        "phpcpd": "vendor/bin/grumphp run --tasks=phpcpd",
        "phpcs": "vendor/bin/grumphp run --tasks=phpcs",
        "phpmd": "vendor/bin/grumphp run --tasks=phpmd",
        "phpstan": "vendor/bin/grumphp run --tasks=phpstan",
        "phpunit": "vendor/bin/phpunit --configuration=phpunit.qa-php.xml"
    }
```

File: .gitignore [hunk 7]
```diff
@@ -12,3 +13,6 @@
 
 # Exclude the cghooks.lock file.
 /cghooks.lock
+
+# QA files.
+/*.qa-php.*
```
Comment (zero2one) [inline .gitignore:19]: # Exclude local confg files.
/*.local.*
/*.qa-php.*

File: composer.json [hunk 8]
```diff
@@ -1,32 +1,36 @@
...
+    ],
     "authors": [
         {
             "name": "Peter Decuyper",
             "email": "peter.decuyper@digipolis.gent"
         }
     ],
-    "minimum-stability": "dev",
-    "prefer-stable": true,
+    "homepage": "https://bitbucket.org/digipolisgent/php_package_gent-services-profile",
```
Comment (zero2one) [inline composer.json]: where hosted on GitHub now (and on district09)

**Review by zero2one — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 7 above)
Comment (MPParsley) [reply .gitignore:19]:
```suggestion
/*.local.*
/*.qa-php.*
```

File: composer.json (hunk 8 above)
Comment (MPParsley) [reply composer.json]:
```suggestion
    "homepage": "https://github.com/StadGent/php_package_services-opening-hours",
```

File: composer.json (hunk 5 above)
Comment (MPParsley) [inline composer.json:21]:
```suggestion
        "digipolisgent/api-client": "^1.1.0",
        "ext-json": "*",
```

File: composer.json (hunk 2 above)
Comment (MPParsley) [inline composer.json]:
```suggestion
```

File: composer.json (hunk 3 above)
Comment (MPParsley) [inline composer.json]:
```suggestion
        "mihaeu/html-formatter": "*"
```

Comment (MPParsley) [inline composer.json:30]:
```suggestion
        "mihaeu/html-formatter": "*"
```

**Pull request merged**